Better boost docs for integrators (#312)
chainflip-io/chainflip-docs: pages/lp/integrations/boost-integration.mdx

---

title: Boost Integration

description: How to integrate boost into your application.

---

import { Callout } from "@/components";

# Boost Integration

This page outlines how to integrate boost into your application.

See here for an [introduction to boost](../../swapping/intro-to-boost.mdx).

For a detailed description of how boost works, see [here](../boost/how-boost-works.mdx).

## Boost Pool Integration

If you wish to allow for people to contribute to boost pools in your application, here is how you can do it.

First, the user needs to have an [on-chain account with the `LiquidityProvider` role](../integrations/chainflip-account/lp.mdx).

The user then needs to deposit funds into the account, so they have a free balance.

There are then two relevant extrinsics for boost.

### `add_boost_funds(asset: Asset, amount: Amount, fee_tier: u16)`

> For more details on how `add_boost_funds` works, see [here](../boost/how-boost-works.mdx#adding-boost-funds).

Reviewer: ```suggestion
For more details on how `add_boost_funds` works, see [here](../boost/how-boost-works.mdx#creating-a-boost-pool-and-adding-liquidity).
```

Author: I split that into two headings in the how-boost-works file, makes more sense that way

---

title: Boost Integration

description: How to integrate boost into your application.

---

import { Callout } from "@/components";

# Boost Integration

This page outlines how to integrate boost into your application.

See here for an [introduction to boost](../../swapping/intro-to-boost.mdx).

For a detailed description of how boost works, see [here](../boost/how-boost-works.mdx).

## Boost Pool Integration

If you wish to allow for people to contribute to boost pools in your application, here is how you can do it.

First, the user needs to have an [on-chain account with the `LiquidityProvider` role](../integrations/chainflip-account/lp.mdx).

The user then needs to deposit funds into the account, so they have a free balance.

There are then two relevant extrinsics for boost.

### `add_boost_funds(asset: Asset, amount: Amount, fee_tier: u16)`

Adds the provided amount from the user's free balance to the boost pool for the given asset and fee tier.

> Note: The `amount` is in atomic units of the currency specified.

Reviewer: we use `smallest unit` instead of `atomic unit` across our docs so lets try to be consistent:

```suggestion
Note: The `amount` should be in the smallest unit of the currency specified.
```

Reviewer: This still says "atomic units", can we change it as suggested above? 🙏

### `stop_boosting(asset: Asset, fee_tier: u16)`

> Removes any of the user's funds from the `available` balance in the boost pool, and queues the pending deposits for withdrawal when they are completed.

Reviewer 2: ```suggestion
Moves any of the user's funds into their free balance, and queues any of their balance that is currently involved in pending deposits to go to their free balance once they are finalized.
```

<Callout type="info">

> If a user calls `add_boost_funds` after calling `stop_boosting`, the funds that were pending confirmation to be withdrawn will be added back to the `available` balance.

Reviewer 2: ```suggestion
If a user calls `add_boost_funds` after calling `stop_boosting`, the funds that were pending confirmation to be withdrawn will be added back to the `available` balance in the boost pool.
```

</Callout>

## How to Use the Extrinsics

To use these extrinsics, you will need to know what parameters to pass to them. You can find the list of available fee tiers for a particular asset using the [Boost Pool Details RPC](../boost/boost-rpcs.mdx#cf_boost_pool_details).

To see all the available assets you can use this RPC:

```bash copy
curl -H "Content-Type: application/json" -d '{
    "id":1,
    "jsonrpc":"2.0",
    "method": "cf_supported_assets",
    "params": []
}' http://localhost:9944
```

> Of these assets, the chain is which pallet instance to interact with e.g. `EthereumIngressEgress` for Ethereum. The asset you then supply to the extrinsics is the asset you wish to use for boosting, or stop boosting, on that particular chain.

Reviewer: Lets use `BitcoinIngressEgress` as an example here so we match the code examples.

Reviewer: This is also showing the old text
